samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -253,15 +254,15 @@ public void waitForFinish() {
   /**
    * Create {@link StreamProcessor} based on {@link StreamApplication} and the config
    * @param config config
-   * @param app {@link StreamApplication}
+   * @param streamGraph {@link StreamGraphImpl}
    * @return {@link StreamProcessor]}
    */
   /* package private */
   StreamProcessor createStreamProcessor(
       Config config,
-      StreamApplication app,
+      StreamGraphImpl streamGraph,
       StreamProcessorLifecycleListener listener) {
-    Object taskFactory = TaskFactoryUtil.createTaskFactory(config, app, new LocalApplicationRunner(config));
+    Object taskFactory = TaskFactoryUtil.createTaskFactory(config, streamGraph);
```

Comment:
I wonder if we can initialize the `StreamApplication` directly in the runner instead of initializing it inside an Util method? It's not obvious to observe that `app#init` is being invoked.

Reply:
+1

Reply:
This is also part of the ApplicationRunner changes that I would love to address in a separate PR. Unfortunately, to remove the multiple invocation of init() per task context, this is the minimum that we will need to get this PR compiled and run.